### PR TITLE
Add a rename macro for (slash) command options.

### PR DESCRIPTION
**Pull request by Bloectasy**

### CODE REVIEW

**Comment by kangalioo** [comment]
Awesome! Can you add an example usage of the new #[rename] parameter attribute to examples/framework_usage/?

File: examples/framework_usage/main.rs [hunk 1]
```diff
@@ -139,12 +138,6 @@ async fn main() {
                 Ok(true)
             })
         }),
-        listener: |_ctx, event, _framework, _data| {
-            Box::pin(async move {
-                println!("Got an event in listener: {:?}", event.event_type());
-                Ok(())
-            })
-        },
```
Comment (kangalioo) [inline examples/framework_usage/main.rs:147]: Why'd you remove this?

File: examples/framework_usage/main.rs [hunk 2]
```diff
@@ -158,12 +151,6 @@ async fn main() {
             })
         })
         .options(options)
-        .client_settings(|b| {
-            b.intents(
-                serenity::GatewayIntents::non_privileged()
-                    | serenity::GatewayIntents::MESSAGE_CONTENT,
-            )
-        })
```
Comment (kangalioo) [inline examples/framework_usage/main.rs:166]: And this

Comment (Bloectasy) [reply examples/framework_usage/main.rs]: Branch conflicted somehow it has been restored in the latest commit, same goes for the listener option.

**Pull request merged**